sound/soc/sof/core.c

```diff
@@ -390,13 +397,6 @@ int snd_sof_device_remove(struct device *dev)
 		snd_sof_free_trace(sdev);
 	}
 
-	/*
-	 * Unregister machine driver. This will unbind the snd_card which
-	 * will remove the component driver and unload the topology
-	 * before freeing the snd_card.
-	 */
-	snd_sof_machine_unregister(sdev, pdata);
-
```

Comment:
I dont think this change is really critical. When unloading topology, we do no send IPCs nor do we power off cores anymore with my PR #3019.

Reply:
It was suggested by @plbossart to move it there while we are here.